### DISCUSSION TITLE
How do I use type: textColor in the addon manifest?

### DISCUSSION THREAD

**Discussion by Samq64** [ScratchAddons/ScratchAddons#3778] [Help] [answered]
I was trying to figure out how to set the text colour to black/white depending on another setting's brightness and noticed that that's already being done in `editor-dark-mode` through custom css variables and `type: textColor`. Could someone please explain how it works?

**Comment by mxmou** [comment, marked as answer]
The `addon.json` should look like this:
```json
{
  "customCssVariables": [
    {
      "name": "exampleTextColor",
      "value": {
        "type": "textColor",
        "source": {
          "type": "settingValue",
          "settingId": "exampleSetting"
        }
      }
    }
  ]
}
```

The `"value"` object can have additional properties:
* `threshold` (integer, 0-255): White text will be used when the setting value is at least as dark as the this value. Default is 170 (equivalent to `#aaaaaa`).
* `black`: The black text color. Default is `#575e75`.
* `white`: The white text color. Default is `#ffffff`.

All of these can also be objects with a `type` attribute, similar to `value` and `source`. This allows things like using another color setting as the threshold.

This sets a CSS variable that can be used in a userstyle like this:
```css
.example-element {
  color: var(--exampleAddon-exampleTextColor); /* the addon name must be camelCase */
}
```

**Reply by Samq64** [reply]
I tried that already before opening the discussion and it didn't work. It turns out I was editing a copy of the CSS file instead of the original.

Thanks for taking the time to answer though!
